### PR TITLE
[chore] add fixes for otelcol-contrib to release pipeline

### PR DESCRIPTION
**Pull request by mowies**
This PR adds missing changes to the release pipeline that should have been part of https://github.com/open-telemetry/opentelemetry-collector-releases/pull/666

### CODE REVIEW

File: .github/workflows/base-release.yaml
Comment (TylerHelmuth) [inline .github/workflows/base-release.yaml:95]: What happens when `release --clean --split --timeout 2h --release-header-tmpl=../../.github/release-template.md` is called again below?

Comment (mowies) [reply .github/workflows/base-release.yaml:95]: the build for otelcol-contrib is split between 2 different goreleaser config files as of #666. so the first call gets a different config than the second one.
the first call will just build the binaries, the second one continues business as is, while importing the prebuilt binaries and it should then continue as usual

**Review by TylerHelmuth — CHANGES_REQUESTED** [review]
Blocking this for now in favor of https://github.com/open-telemetry/opentelemetry-collector-releases/pull/761.  @mowies to move forward with the release in a guaranteed way we are gonna revert the original breaking PR.  We'll get you a goreleaser-pro key so you can test everything in your fork before we merge it back into this repo

**Comment by mowies** [comment]
sounds good, I was gonna suggest the same thing

**Comment by mowies** [comment]
closing this in favor of https://github.com/open-telemetry/opentelemetry-collector-releases/pull/762

**Comment by mowies** [comment]
@TylerHelmuth @atoulme @mx-psi the new PR is ready for review. I was able to properly test it with a goreleaser-pro token:
https://github.com/open-telemetry/opentelemetry-collector-releases/pull/762

Test release on my fork was successful, i linked the workflow runs and the test release in the PR

**Pull request closed without merging**
